model/aktivita.php

```diff
@@ -2360,4 +2360,18 @@ public static function moznaHlavniAktivitaPodleUrl(string $url, int $rok, int $t
         }
         return static::zId($idHlavniAktivity);
     }
+    
+    /**
+     * Všechny roky ve kterých se aktivity konaly
+     * @return int[]
+     */
+    public static function roky() {
+        $roky = [];
+        $o = dbQuery('SELECT DISTINCT rok FROM `akce_seznam` WHERE rok IS NOT NULL AND rok > 0 ORDER BY rok DESC');
```

Comment:
Na tohle máme funkci `dbOneArray` (já ty názvy nevymýšlel .-)

Takže by to mohlo být nějak jako

```php
return array_map('intval', dbOneArray('SELECT DISTINCT rok FROM `akce_seznam` WHERE rok IS NOT NULL AND rok > 0 ORDER BY rok DESC'));
```

Jinak by to klidně taky šlo bez SQL dotazu, nějak jako
```php
$roky = range(2009, ROK);
```
Přičemž v roce 2020 Gamecon nebyl kvůli covidu a to i ten SQL dotaz neprozradí (nějaké aktivity připravené byly).

Takže buďto je to jedno a rok 2020 zahrneme, nebo ho ručně ze seznamu vyřadíme
```php
$indexCovidRoku = array_search(2020, $roky);
unset($roky[$indexCovidRoku]);
```
nebo upravíme ten SQL o stav aktivity "jiná než jenom nová"
```sql
SELECT DISTINCT rok
FROM akce_seznam
WHERE rok IS NOT NULL AND rok > 0 AND akce_seznam.stav != 0
ORDER BY rok DESC
```

Reply:
zvolil jsem teda fixní verzi a dal jsem to rovnou do programu když se to jinde nepoužívá